packages/xo-web/src/xo-app/xoa/update/index.js

```diff
@@ -84,25 +89,52 @@ const Updates = decorate([
     initialState: () => ({
       ...initialProxyState(),
       ...initialRegistrationState(),
+      ...initialChannels(),
       askRegisterAgain: false,
       showPackagesList: false,
     }),
     effects: {
       async configure() {
-        await xoaUpdater.configure(
-          pick(this.state, [
+        const { effects, state } = this
+        await xoaUpdater.configure({
+          ...pick(state, [
             'proxyHost',
             'proxyPassword',
             'proxyPort',
             'proxyUser',
-          ])
-        )
-        return this.effects.resetProxyConfig()
+          ]),
+          channel: state.isUnlistedChannel
+            ? state.unlistedChannel
+            : state.channel,
+        })
+        await Promise.all([effects.resetProxyConfig(), effects.update()])
       },
-      initialize() {
-        return this.effects.update()
+      async initialize() {
+        await Promise.all([
+          this.effects.initializeChannels(),
+          this.effects.update(),
+        ])
+      },
+      async initializeChannels() {
+        const { state } = this
+
+        const channels = await xoaUpdater.getReleaseChannels()
+        state.channels = channels
+
+        const { channel } = this.props.xoaConfiguration
+        if (channel !== undefined) {
+          const isPublicChannel = channel in channels
+
+          if (isPublicChannel) {
+            state.channel = channel
+          } else {
+            state.channel = UNLISTED_CHANNEL_VALUE
+            state.unlistedChannel = channel
```

Comment:
Instead of declaring two variables, you can add a `boolean` to the options which indicate if it's public or not.

Reply:
I don't think that will help a lot, because we need a special value for the `Select`.

Reply:
The channel value is kept, we send it even if it's a private channel. So we can create a computed for `channel`. This computed can take a value from state or from props.